android/app/src/main/kotlin/org/getlantern/lantern/service/LanternService.kt

```diff
@@ -190,4 +132,28 @@ open class LanternService : Service(), Runnable {
             .setClass(this, AutoStarter::class.java)
         sendBroadcast(broadcastIntent)
     }
+
+    private class CreateUser(val service: LanternService) : Runnable {
+        private var attempts: Int = 0
+
+        override fun run() {
+            try {
+                val userCreated = LanternApp.getSession().createUser()
+                if (userCreated) {
+                    service.createUserHandler.removeCallbacks(service.createUserRunnable)
+                    EventHandler.postStatusEvent(LanternStatus(Status.ON))
+                    EventHandler.postAccountInitializationStatus(AccountInitializationStatus.Status.SUCCESS)
+                }
+            } catch (e: Exception) {
+                if (attempts >= MAX_CREATE_USER_TRIES) {
+                    Logger.error(TAG, "Max. number of tries made to create Pro user")
+                    EventHandler.postAccountInitializationStatus(AccountInitializationStatus.Status.FAILURE)
+                    return
+                }
+                attempts++
+                service.createUser(attempts)
+            }
+        }
+    }
```

Comment:
We don't have to do it on this PR but we could probably get rid of this Runnable and just create the user on the Go side (similar to the desktop code) [here](https://github.com/getlantern/lantern-client/blob/main/internalsdk/android.go#L423)

Reply:
I agree. Let's keep same for now.